File: Selenium/Project/Activity09.java
package project;

import java.time.Duration;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

public class Activity09 {
	WebDriver driver;

	@BeforeClass
	public void login() {
		driver = new FirefoxDriver();
		driver.get("https://alchemy.hguy.co/crm/index.php?action=Login&module=Users");
		driver.findElement(By.id("user_name")).sendKeys("admin");
		driver.findElement(By.id("username_password")).sendKeys("pa$$w0rd");
		driver.findElement(By.id("bigbutton")).click();

	}

	@Test
	public void traversingTables2() {

		WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
		Actions action = new Actions(driver);
		wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("grouptab_0")));
		WebElement salatesTab = driver.findElement(By.id("grouptab_0"));
		action.moveToElement(salatesTab).build().perform();
		wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("moduleTab_9_Leads")));
		driver.findElement(By.id("moduleTab_9_Leads")).click();

		wait.until(ExpectedConditions.visibilityOfElementLocated(
				By.xpath("//*[@class='list-view-rounded-corners']//tbody[1]/tr[contains(@class,'oddListRow')][1]")));
		List<WebElement> names = driver.findElements(By.xpath("//td[@type='name']//a"));
		List<WebElement> user = driver.findElements(By.xpath("//td[@type='relate']//a"));

		for (int i = 0; i < 10; i++) {
			System.out.println("Name = " + names.get(i).getText() + " , " + "User = " + user.get(i).getText());
		}
	}

	@AfterClass

	public void closeBrowser() {
		driver.quit();
	}
}

File: Selenium/Project/Activity09.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.common.action_chains import ActionChains

with webdriver.Firefox() as driver:

    driver.get("https://alchemy.hguy.co/crm/index.php?action=Login&module=Users")

    username = driver.find_element(By.ID, "user_name")
    password = driver.find_element(By.ID, "username_password")
    username.send_keys("admin")
    password.send_keys("pa$$w0rd")
    driver.find_element(By.ID, "bigbutton").click()

    wait = WebDriverWait(driver, 10)
    wait.until(EC.visibility_of_element_located((By.ID, "grouptab_0")))
    act = ActionChains(driver)

    act.move_to_element(driver.find_element(By.ID, "grouptab_0")).perform()

    wait.until(EC.visibility_of_element_located((By.ID, "moduleTab_9_Leads")))

    driver.find_element(By.ID, "moduleTab_9_Leads").click()

    print("User navigated to Sales > Leads ")

    wait.until(
        EC.visibility_of_element_located(
            (
                By.XPATH,
                "//*[@class='list-view-rounded-corners']//tbody[1]/tr[contains(@class,'oddListRow')][1]",
            )
        )
    )

    listOfnames = driver.find_elements(
        By.XPATH,
        "//td[@type='name']//a",
    )

    listOfusers = driver.find_elements(
        By.XPATH,
        "//td[@type='relate']//a",
    )

    rangeUpTo = 1
    for name, user in zip(listOfnames, listOfusers):

        if rangeUpTo <= 10:
            print("Name = ", name.text, " , ", "User = " + user.text)
            rangeUpTo += 1

    driver.quit()
